### ISSUE TITLE
No assets.h file is generated after running prepare-game-assets

### ISSUE THREAD

**Issue by tfinnegan937** [matthewcpp/framework64#3]
I am attempting to build the simple_scene demo. I am building on Windows 11.

To do this I:

1. Created a new game using the quickstart guide in the git repository "mygame".
2. Followed the N64 VSCode build guide to get the dev container set up
3. Copied all of the assets from the repo into my mygame/assets folder
4. Copied the assets.json manifest file, source files, and CMakeLists.txt from the simple_scene demo into my mygame/assets folder
5. Copied the "fire.c" and "fire.h" files from the billboard example into my src/ folder, and updated CMakeLists.txt accordingly.
6. Ran the command "npm run prepare-game-assets n64" in mygame/
7. Attempted to build the project.

I get the following output:

```bash
[main] Building folder: mygame 
[main] Configuring project: mygame 
[proc] Executing command: /usr/bin/cmake --no-warn-unused-cli -DCMAKE_BUILD_TYPE:STRING=Release -DFW64_PLATFORM_N64:STRING=ON -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -S/workspaces/mygame -B/workspaces/mygame/build_n64 -G "Unix Makefiles"
[cmake] Not searching for unused variables given on the command line.
[cmake] Configuring for Platform: N64-libultra
[cmake] -- Configuring done
[cmake] -- Generating done
[cmake] -- Build files have been written to: /workspaces/mygame/build_n64
[build] Starting build
[proc] Executing command: /usr/bin/cmake --build /workspaces/mygame/build_n64 --config Release --target all -j 18 --
[build] [ 82%] Built target framework64
[build] [ 83%] Building C object src/CMakeFiles/simple_scene.dir/flame.c.obj
[build] [ 87%] Building C object src/CMakeFiles/simple_scene.dir/ui.c.obj
[build] [ 87%] Building C object src/CMakeFiles/simple_scene.dir/game.c.obj
[build] [ 89%] Building C object src/CMakeFiles/simple_scene.dir/__/lib/framework64/src/n64/main_n64.c.obj
[build] [ 91%] Building C object src/CMakeFiles/simple_scene.dir/asm/asset_data.s.obj
[build] /workspaces/mygame/build_n64/src/asm/asset_data.s: Assembler messages:
[build] /workspaces/mygame/build_n64/src/asm/asset_data.s:7: Error: file not found: /workspaces/mygame/build_n64/bin/simple_scene/assets/assets.dat
[build] make[2]: *** [src/CMakeFiles/simple_scene.dir/build.make:167: src/CMakeFiles/simple_scene.dir/asm/asset_data.s.obj] Error 1
[build] make[2]: *** Waiting for unfinished jobs....
[build] /workspaces/mygame/src/flame.c:4:10: fatal error: assets/assets.h: No such file or directory
[build]     4 | #include "assets/assets.h"
[build]       |          ^~~~~~~~~~~~~~~~~
[build] compilation terminated.
[build] make[2]: *** [src/CMakeFiles/simple_scene.dir/build.make:63: src/CMakeFiles/simple_scene.dir/flame.c.obj] Error 1
[build] /workspaces/mygame/src/ui.c:3:10: fatal error: assets/assets.h: No such file or directory
[build]     3 | #include "assets/assets.h"
[build]       |          ^~~~~~~~~~~~~~~~~
[build] compilation terminated.
[build] /workspaces/mygame/src/game.c:3:10: fatal error: assets/assets.h: No such file or directory
[build]     3 | #include "assets/assets.h"
[build]       |          ^~~~~~~~~~~~~~~~~
[build] make[2]: *** [src/CMakeFiles/simple_scene.dir/build.make:102: src/CMakeFiles/simple_scene.dir/ui.c.obj] Error 1
[build] compilation terminated.
[build] make[2]: *** [src/CMakeFiles/simple_scene.dir/build.make:115: src/CMakeFiles/simple_scene.dir/game.c.obj] Error 1
[build] /workspaces/mygame/lib/framework64/src/n64/main_n64.c:4:10: fatal error: assets/assets.h: No such file or directory
[build]     4 | #include "assets/assets.h"
[build]       |          ^~~~~~~~~~~~~~~~~
[build] compilation terminated.
[build] make[2]: *** [src/CMakeFiles/simple_scene.dir/build.make:128: src/CMakeFiles/simple_scene.dir/__/lib/framework64/src/n64/main_n64.c.obj] Error 1
[build] make[1]: *** [CMakeFiles/Makefile2:180: src/CMakeFiles/simple_scene.dir/all] Error 2
[build] make: *** [Makefile:84: all] Error 2
[proc] The command: /usr/bin/cmake --build /workspaces/mygame/build_n64 --config Release --target all -j 18 -- exited with code: 2
[driver] Build completed: 00:00:03.006
[build] Build finished with exit code 2
```

I have verified that no assets.h file gets created anywhere in the repository.

The following is the output to my "npm run prepare-game-assets n64", which completes without error:

```bash
PS C:\Users\Tim\mygame> npm run prepare-game-assets n64

> mygame@0.0.1 prepare-game-assets
> node lib/framework64/scripts/PrepareGameAssets.js n64

No target specified. using default target: mygame
Processing Mesh: penguin/penguin.gltf
Processing Image Atlas: fire_sprite
Processing Font: Consolas.ttf
Processing Level: simple_scene/simple_scene.gltf
```

Here is my assets.json:

```json
{
    "fonts": [
        {
            "src": "Consolas.ttf",
            "size": 12,
            "name": "Consolas12"
        }
    ],
    "images": [
        {
            "name": "fire_sprite",
            "frameDir": "fire_sprite",
            "hslices": 10,
            "vslices": 1,
            "frameSize": "128x128",
            "resize": "320x32",
            "format": "rgba32"
        }
    ],
    "levels": [
        {
            "src": "simple_scene/simple_scene.gltf",
            "typeMap": 0,
            "layerMap": 0
        }
    ],
    "meshes": [
        {
            "src": "penguin/penguin.gltf"
        }
    ],
    "layers": {
        "default": 1,
        "ground": 2,
        "wall": 4
    }
}
```

And finally, here is my src/CMakelists.txt 

```cmake
set(src_files
    flame.h flame.c
    chase_cam.h chase_cam.c
    player.h player.c
    ui.h ui.c
    game.h game.c )

    create_game(
        TARGET simple_scene
        SOURCES ${src_files})
```

**Comment by matthewcpp** [comment]
Thanks for providing detailed reproduction steps.  I followed along and I was able to get the ROM to build correctly by making one small tweak to the cmake lists file.  I think that when you copied over the cmake file from the example you forgot to update the TARGET from `simple_scene` to `mygame`.  After making that change I was able to get it to build and run (tested in ares).

If you end up trying to use this for the current game jam, feel free to reach out to me on discord.  My user name is: spirit.of.76
